src/cms/contentful.ts

```diff
@@ -84,7 +84,9 @@ export class Contentful extends AbstractAgnosticCMSHarmonizerClient {
         }, {});
 
       case ContentfulResourceType.ASSET:
-        return `https:${item.fields?.file?.url}`;
+        return item.fields?.file?.url ?
+            `https:${item.fields?.file?.url}`
+          : undefined;
```

Comment:
Wouldn't be better to return `null` instead? This can cause unintentional and hard to debug cases. For example when using the service in `getServerSideProps` where it is forbidden to return `undefined`.

Reply:
I'm afraid yes. So there must be no `undefined` in the response payload _at all_ otherwise it wouldn't be serialisable:

![src](https://p266.p3.n0.cdn.zight.com/items/Jru5O1pG/3c88284c-6158-47c1-8baf-2cd7f8e871fc.jpg?v=8fa3eaae1c21c37eb517278026d97ed9)